static/src/javascripts/projects/common/modules/commercial/braze/brazeBanner.tsx

```diff
@@ -81,20 +91,29 @@ const getMessageFromUrlFragment = () => {
 			try {
 				const dataFromBraze = JSON.parse(
 					decodeURIComponent(forcedMessage),
-				);
-				return {
-					extras: dataFromBraze,
-					logImpression: () => {},
-					logButtonClick: () => {},
-				};
+				) as unknown;
+
+				if (isExtrasData(dataFromBraze)) {
+					return {
+						extras: dataFromBraze,
+						logImpression: () => {
+							return;
+						},
+						logButtonClick: () => {
+							return;
+						},
+					};
+				}
 			} catch (e) {
 				// Parsing failed. Log a message and fall through.
-				console.log(`There was an error with ${key}:`, e.message);
+				if (e instanceof Error) {
```

Comment:
Technically, if it’s a JSON parsing error, you can check if `instanceof SyntaxError`.

Reply:
Oh interesting, thanks. I wonder if it's better to keep this general (and therefore future-proof if other changes are introduced to the `try` block) or to use the narrower type. I think since all we're doing is logging, general seems okay.